langchain/vectorstores/rocksetdb.py

```diff
@@ -0,0 +1,308 @@
+            embedding_key: column in Rockset collection to use to store the embedding.
+                           Note: We must apply `VECTOR_ENFORCE()` on this column via
+                           Rockset ingest transformation.
+
+        """
+        if not isinstance(client, RocksetClient):
+            raise ValueError(
+                f"client should be an instance of rockset.RocksetClient, "
+                f"got {type(client)}"
+            )
+        # TODO: check that `collection_name` exists in rockset. Create if not.
+        self._client = client
+        self._collection_name = collection_name
+        self._embeddings = embeddings
+        self._text_key = text_key
+        self._embedding_key = embedding_key
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        batch_size: int = 32,
+    ) -> List[str]:
+        """Run more texts through the embeddings and add to the vectorstore
+
+                Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+            ids: Optional list of ids to associate with the texts.
+            batch_size: Send documents in batches to rockset.
+
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+
+        """
+        batch = []
+        stored_ids = []
+
+        for i, text in enumerate(texts):
+            if len(batch) == batch_size:
+                stored_ids += self._write_documents_to_rockset(batch)
+                batch = []
+            doc = {}
+            if metadatas and len(metadatas) > i:
+                doc = metadatas[i]
+            if ids and len(ids) > i:
+                doc["_id"] = ids[i]
+            doc[self._text_key] = text
+            doc[self._embedding_key] = self._embeddings.embed_query(text)
+            batch.append(doc)
+        if len(batch) > 0:
+            stored_ids += self._write_documents_to_rockset(batch)
+            batch = []
+        return stored_ids
+
+    @classmethod
+    def from_texts(
+        cls,
+        texts: Iterable[str],
+        embeddings: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        client: Any = None,
+        collection_name: str = "",
+        text_key: str = "",
+        embedding_key: str = "",
+        ids: Optional[List[str]] = None,
+        batch_size: int = 32,
+    ) -> RocksetDB:
+        """Create Rockset wrapper with existing texts.
+        This is intended as a quicker way to get started.
+        """
+
+        # Sanitize imputs
+        assert client is not None, "Rockset Client cannot be None"
+        assert collection_name != "", "Collection name cannot be empty"
+        assert text_key != "", "Text key name cannot be empty"
+        assert embedding_key != "", "Embedding key cannot be empty"
```

Comment:
For these strings, I think you can just do `assert var_name`. It has the advantage of also checking None vars since they could still explicitly pass a None in.